src/test/regress/expected/node_conninfo_reload.out

```diff
@@ -0,0 +1,408 @@
+-- Make sure changes citus.node_conninfo shutdown connections with old settings
+CREATE SCHEMA node_conninfo_reload;
+SET search_path TO node_conninfo_reload;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+create table test(a int);
+select create_distributed_table('test', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- Make sure a connection is opened and cached
+select count(*) from test where a = 0;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+show citus.node_conninfo;
+ citus.node_conninfo
+---------------------------------------------------------------------
+ sslmode=require
+(1 row)
+
+-- Set sslmode to something that does not work when connecting
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=doesnotexist';
+select pg_reload_conf();
+ pg_reload_conf
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select pg_sleep(0.1); -- wait for config reload to apply
+ pg_sleep
+---------------------------------------------------------------------
+
+(1 row)
+
+show citus.node_conninfo;
+ citus.node_conninfo
+---------------------------------------------------------------------
+ sslmode=doesnotexist
+(1 row)
+
+-- Should give a connection error because of bad sslmode
+select count(*) from test where a = 0;
+ERROR:  connection error: localhost:xxxxx
+DETAIL:  invalid sslmode value: "doesnotexist"
+-- Reset it again
+ALTER SYSTEM RESET citus.node_conninfo;
+select pg_reload_conf();
+ pg_reload_conf
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select pg_sleep(0.1); -- wait for config reload to apply
+ pg_sleep
+---------------------------------------------------------------------
+
+(1 row)
+
+show citus.node_conninfo;
+ citus.node_conninfo
+---------------------------------------------------------------------
+ sslmode=require
+(1 row)
+
+-- Should work again
+select count(*) from test where a = 0;
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=doesnotexist';
+BEGIN;
+-- Should still work (no SIGHUP yet);
+select count(*) from test where a = 0;
```

Comment:
I think this is not guaranteed to work, so could lead to some random test failures. 

In the previous `SELECT` command, the executor might have opened, say 2 connections per node. And, in this statement, due to the load in the cluster, it might require 3 connections. And, the 3rd connection might fail.

To make this test file more robust to such random failures, I suggest we use `SET citus.max_adaptive_executor_pool_size TO 1` so that we ensure that we'll have a single connection.